modules/storages/app/workers/storages/manage_nextcloud_integration_events_job.rb

```diff
@@ -30,22 +30,20 @@ module Storages
   class ManageNextcloudIntegrationEventsJob < ApplicationJob
     include ManageNextcloudIntegrationJobMixin
 
-    DEBOUNCE_TIME = 5.seconds.freeze
-    THREAD_KEY = :manage_nextcloud_integration_events_job_debounce_happend_at
+    SINGLE_THREAD_DEBOUNCE_TIME = 4.seconds.freeze
+    MULTI_THREAD_DEBOUNCE_TIME = 5.seconds.freeze
+    KEY = :manage_nextcloud_integration_events_job_debounce_happend_at
 
     queue_with_priority :above_normal
 
     def self.debounce
-      last_debounce_happend_at = Thread.current[THREAD_KEY]
-      if last_debounce_happend_at.blank? || Time.current > (last_debounce_happend_at + DEBOUNCE_TIME)
-        count = Delayed::Job
-                  .where("handler LIKE ?", "%job_class: #{self}%")
-                  .where(locked_at: nil)
-                  .where('run_at <= ?', DEBOUNCE_TIME.from_now)
-                  .delete_all
-        Rails.logger.info("deleted: #{count} jobs")
-        Thread.current[THREAD_KEY] = Time.current
-        set(wait: DEBOUNCE_TIME).perform_later
+      last_current_thread_debounce_happend_at = RequestStore.store[KEY]
+      if last_current_thread_debounce_happend_at.blank? ||
```

Comment:
@ba1ash 

Could we please put these two lines (40 and 41) into a method. And as much as I like descriptive variable names I think in a separate method we could get away with a shorter variable name for it and give the method the descriptive name instead.

Reply:
Yes, sure.